var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -182,19 +191,13 @@ def _find_mkl_libs(self, shared):
         libs.append("libmkl_core")
 
         if self.spec.satisfies("+cluster"):
-            if any(
-                self.spec.satisfies(m)
-                for m in ["^intel-oneapi-mpi", "^intel-mpi", "^mpich", "^cray-mpich"]
-            ):
+            if self.spec.satisfies("mpi=mpich"):
```

Comment:
After some more thought:
this change will break the package for some existing users. Could you instead leave line 187 and add "mpi=mpich" to the list there? Same for "mpi=openmpi" below